app/src/main/kotlin/com/wire/android/ui/home/conversations/model/MessageTypes.kt

```diff
@@ -155,23 +170,55 @@ internal fun MessageAsset(assetName: String, assetExtension: String, assetSizeIn
                     color = MaterialTheme.wireColorScheme.secondaryText,
                     style = MaterialTheme.wireTypography.subline01
                 )
-                Text(
-                    modifier = Modifier
-                        .padding(start = MaterialTheme.wireDimensions.spacing8x)
-                        .constrainAs(downloadStatus) {
-                            top.linkTo(parent.top)
-                            end.linkTo(parent.end)
-                            bottom.linkTo(parent.bottom)
+                Row(
+                    modifier = Modifier.wrapContentWidth().constrainAs(downloadStatus) {
+                        top.linkTo(parent.top)
+                        end.linkTo(parent.end)
+                        bottom.linkTo(parent.bottom)
+                    },
+                ) {
+                    Text(
+                        modifier = Modifier.padding(end = dimensions().spacing4x),
+                        text = getDownloadStatusText(assetDownloadStatus),
+                        color = MaterialTheme.wireColorScheme.run {
+                            if (assetDownloadStatus == FAILED) error else secondaryText
                         },
-                    text = stringResource(R.string.asset_message_download_text),
-                    color = MaterialTheme.wireColorScheme.secondaryText,
-                    style = MaterialTheme.wireTypography.subline01
-                )
+                        style = MaterialTheme.wireTypography.subline01
+                    )
+                    addExtraDownloadStatusIcon(assetDownloadStatus)
```

Comment:
I think we can just do it the Compose way and call it DownloadStatusIcon() what you think ?